### PR TITLE
 handle duplicates and other errors when submitting an antibody #86

### PR DESCRIPTION
**Pull request by Anto8819**
``
Screenshot 2022-09-27 at 12 32 58
Screenshot 2022-09-27 at 12 31 24

### CODE REVIEW

File: applications/portal/frontend/src/components/Submit/CommercialForm.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { AlertIcon } from "../icons";
 import StepNavigation from "./StepNavigation";
 import { useFormik } from "formik";
 import * as yup from "yup";
-import { addAntibody } from "../../services/AntibodiesService";
+import { postNewAntibody } from "../../utils/antibody";
```
Comment (filippomc) [inline applications/portal/frontend/src/components/Submit/CommercialForm.tsx:22]: why the logic is for the post has been moved from the service to utils?

Comment (filippomc) [reply applications/portal/frontend/src/components/Submit/CommercialForm.tsx:22]: utils should be pure functions

Comment (Anto8819) [reply applications/portal/frontend/src/components/Submit/CommercialForm.tsx]: moved to helpers

File: applications/portal/frontend/src/utils/antibody.ts [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+import { addAntibody } from "../services/AntibodiesService";
 
 export function getProperCitation(a: Antibody) {
-  return `(${a.vendorName}#${a.catalogNum}, RRID:AB_${a.abId})`
-}
\ No newline at end of file
+  return `(${a.vendorName}#${a.catalogNum}, RRID:AB_${a.abId})`;
+}
+
+export function postNewAntibody(a: Antibody, props) {
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/portal/frontend/src/utils/antibody.ts]: moved to helpers

**Review by filippomc — APPROVED** [review]
Looks good to me

**Pull request merged**